### PR TITLE
Remove ApiVersion Parameter and change Get-AzResourceGroupOperation implementation to use SDK

### PR DESCRIPTION
**Pull request by Xynoclafe**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
For public API changes to cmdlets, a cmdlet design review was approved for the changes [in this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr)

Besides, removing a parameter is a breaking change. The nearest sprint open for breaking changes will be [S177](https://github.com/Azure/azure-powershell/milestones). We suggest:

1. Submit a design review and get it passed
1. Add [breaking change attributes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md) to notify our customers in advance
1. In S177, submit the PR to make breaking change

**Comment by Xynoclafe** [comment]
> For public API changes to cmdlets, a cmdlet design review was approved for the changes [in this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
> 
> Besides, removing a parameter is a **breaking change**. The nearest sprint open for breaking changes will be [S177](https://github.com/Azure/azure-powershell/milestones). We suggest:
> 
> 1. Submit a design review and get it passed
> 2. Add [breaking change attributes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md) to notify our customers in advance
> 3. In S177, submit the PR to make breaking change

Breaking change attributes have already been added and are present in the Master branch
Also, the PR in S177 would be made into Master and not into Az.Resources-preview branch right?

**Comment by isra-fel** [comment]
> Breaking change attributes have already been added and are present in the Master branch

Great

> Also, the PR in S177 would be made into Master and not into Az.Resources-preview branch right?

Ah yes, but still please submit a design review. Thanks

File: src/Resources/ResourceManager/Implementation/ResourceGroupDeployments/GetAzureResourceGroupDeploymentOperationCmdlet.cs [hunk 1]
```diff
@@ -41,6 +38,7 @@ public class GetAzureResourceGroupDeploymentOperationCmdlet : ResourceManagerCmd
         /// <summary>
         /// Gets or sets the subscription id parameter.
         /// </summary>
+        /// [CmdletParameterBreakingChange("SubscriptionId", ChangeDescription = "Parameter is deprecated and is a no-op parameter. It will be retired in a future release")]
         [Parameter(Mandatory = false, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The subscription to use.")]
         [ValidateNotNullOrEmpty]
         public Guid? SubscriptionId { get; set; }
```
Comment (filizt) [inline src/Resources/ResourceManager/Implementation/ResourceGroupDeployments/GetAzureResourceGroupDeploymentOperationCmdlet.cs:44]: > public Guid? SubscriptionId { get; set; } [](start = 7, length = 42)

If we're not using SubscriptionId parameter anymore let's remove it.

Comment (filizt) [inline src/Resources/ResourceManager/Implementation/ResourceGroupDeployments/GetAzureResourceGroupDeploymentOperationCmdlet.cs:44]: >public Guid? SubscriptionId { get; set; } [](start = 8, length = 41)

Let's add a task on the board to remove this parameter later.

**Comment by Xynoclafe** [comment]
> Ah yes, but still please submit a design review. Thanks
Submitted a design review https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/707

**Comment by msJinLei** [comment]
@Xynoclafe You need to suppress the breaking change warning in your PR to pass the static analysis.

File: src/Resources/Resources/help/Get-AzResource.md [hunk 2]
```diff
@@ -216,7 +216,7 @@ Required: False
 Position: Named
 Default value: None
 Accept pipeline input: False
-Accept wildcard characters: True
+Accept wildcard characters: False
```
Comment (filizt) [inline src/Resources/Resources/help/Get-AzResource.md:219]: Not sure if these are related to your changes.

**Comment by Xynoclafe** [comment]
> @Xynoclafe You need to suppress the breaking change warning in your PR to pass the static analysis.
Working on it

**Comment by Xynoclafe** [comment]
@msJinLei @isra-fel
What do I do about false positive static analysis errors? I updated the BreakingChangesIssues.csv file with all the issues that should arise from removal of ApiVersion, but the analysis still throws errors saying I've removed multiple parameter sets, which I have not. These issues go away on just adding the parameter back, so they are indeed false positives.
Should I suppress the other errors as well using the csv file? I am not too sure since that is not the error I am expecting with this change.

**Comment by msJinLei** [comment]
> @msJinLei @isra-fel
> What do I do about false positive static analysis errors? I updated the BreakingChangesIssues.csv file with all the issues that should arise from removal of ApiVersion, but the analysis still throws errors saying I've removed multiple parameter sets, which I have not. These issues go away on just adding the parameter back, so they are indeed false positives.
> Should I suppress the other errors as well using the csv file? I am not too sure since that is not the error I am expecting with this change.

We will check the errors since there are unexpected errors

**Comment by Xynoclafe** [comment]
> > @msJinLei @isra-fel
> > What do I do about false positive static analysis errors? I updated the BreakingChangesIssues.csv file with all the issues that should arise from removal of ApiVersion, but the analysis still throws errors saying I've removed multiple parameter sets, which I have not. These issues go away on just adding the parameter back, so they are indeed false positives.
> > Should I suppress the other errors as well using the csv file? I am not too sure since that is not the error I am expecting with this change.
> 
> We will check the errors since there are unexpected ones

Any updates on this?

**Comment by msJinLei** [comment]
> > > @msJinLei @isra-fel
> > > What do I do about false positive static analysis errors? I updated the BreakingChangesIssues.csv file with all the issues that should arise from removal of ApiVersion, but the analysis still throws errors saying I've removed multiple parameter sets, which I have not. These issues go away on just adding the parameter back, so they are indeed false positives.
> > > Should I suppress the other errors as well using the csv file? I am not too sure since that is not the error I am expecting with this change.
> > 
> > 
> > We will check the errors since there are unexpected ones
> 
> Any updates on this?

It is not false positives errors.

When a parameter is removed from cmdlet, it is evaluated twice in static analysis. 
The first time,  we check all the parameters across different parameter sets per cmdlet to see whether a parameter is removed.  If true, report breaking changes. This is what i have suppressed.
The second time,  we check the parameters by each parameter set of cmdlet.  If true, the breaking change is reported by parameter set. The problem here is the error message is always  `The parameter set '{0}' for cmdlet '{1}' has been removed.` which is incomprehensible here.

And so please check the cmdlets and parameter sets reported to see whether there is any change in the parameter set, including removed parameters (optional or mandatory), additional mandatory parameters. If yes, just suppress the error.

Sorry for the trouble, we will improve the tools in the future.

**Comment by Xynoclafe** [comment]
I think I've already regenerated the help files. But yes, I will verify the case with parameter sets and suppress them accordingly

**Review by filizt — APPROVED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**